state/immutable-trie/trie.go

```diff
@@ -119,89 +118,84 @@ var accountArenaPool fastrlp.ArenaPool
 var stateArenaPool fastrlp.ArenaPool // TODO, Remove once we do update in fastrlp
 
 func (t *Trie) Commit(objs []*state.Object) (state.Snapshot, []byte) {
```

Comment:
Is now a good time to return an error and do error handling?
Never-ending TODOs and panics have been a nightmare for a long time.